challenges/2022-04-26-dojo-anniversary-edition/solutions/javascript/matthewfee/Level4.js

```diff
@@ -0,0 +1,303 @@
+class Block {
+  constructor(startPosition) {
+    this.startPosition = startPosition
+    this.gridPosition = startPosition
+    this.isDownVertically = null
+    this.isDownHorizontally = null
+    this.isDown = null
+    this.moveHistory = []
+    this.destination = null
+    this.xDistance = null
+    this.yDistance = null
+    this.updateOrientation()
+  }
+
+  moveUp() {
+    const y = this.gridPosition[0][0]
+    const x = this.gridPosition[0][1]
+
+    if (!this.isDown) {
+      this.gridPosition = [
+        [y - 2, x],
+        [y - 1, x],
+      ]
+    }
+
+    if (this.isDown) {
+      if (this.isDownVertically) {
+        this.gridPosition = [[y - 1, x]]
+      }
+
+      if (this.isDownHorizontally) {
+        this.gridPosition = [
+          [y - 1, x],
+          [y - 1, x + 1],
+        ]
+      }
+    }
+
+    this.moveHistory.push("U")
+    this.updateOrientation()
+  }
+
+  moveDown() {
+    const y = this.gridPosition[0][0]
+    const x = this.gridPosition[0][1]
+
+    if (!this.isDown) {
+      this.gridPosition = [
+        [y + 1, x],
+        [y + 2, x],
+      ]
+    }
+
+    if (this.isDown) {
+      if (this.isDownVertically) {
+        this.gridPosition = [[y + 2, x]]
+      }
+
+      if (this.isDownHorizontally) {
+        this.gridPosition = [
+          [y + 1, x],
+          [y + 1, x + 1],
+        ]
+      }
+    }
+
+    this.moveHistory.push("D")
+    this.updateOrientation()
+  }
+
+  moveRight() {
+    const y = this.gridPosition[0][0]
+    const x = this.gridPosition[0][1]
+
+    if (!this.isDown) {
+      this.gridPosition = [
+        [y, x + 1],
+        [y, x + 2],
+      ]
+    }
+
+    if (this.isDown) {
+      const y = this.gridPosition[0][0]
+      const x = this.gridPosition[0][1]
+
+      if (this.isDownVertically) {
+        this.gridPosition = [
+          [y, x + 1],
+          [y + 1, x + 1],
+        ]
+      }
+
+      if (this.isDownHorizontally) {
+        this.gridPosition = [[y, x + 2]]
+      }
+    }
+
+    this.moveHistory.push("R")
+    this.updateOrientation()
+  }
+
+  moveLeft() {
+    const y = this.gridPosition[0][0]
+    const x = this.gridPosition[0][1]
+    if (!this.isDown) {
+      this.gridPosition = [
+        [y, x - 2],
+        [y, x - 1],
+      ]
+    }
+
+    if (this.isDown) {
+      if (this.isDownVertically) {
+        this.gridPosition = [
+          [y, x - 1],
+          [y + 1, x - 1],
+        ]
+      }
+
+      if (this.isDownHorizontally) {
+        this.gridPosition = [[y, x - 1]]
+      }
+    }
+
+    this.moveHistory.push("L")
+    this.updateOrientation()
+  }
+
+  updateOrientation() {
+    const hasMultipleGridValues = this.gridPosition.length === 2 ? true : false
+
+    this.isDown = hasMultipleGridValues ? true : false
+
+    let xValuesMatch = false
+    let yValuesMatch = false
+
+    if (hasMultipleGridValues) {
+      xValuesMatch =
+        this.gridPosition[0][1] === this.gridPosition[1][1] ? true : false
```

Comment:
I don't think you need a ternary here. From what I understand of js, the ternary syntax is:
```
(expression) ? <if true use this value> : <otherwise use this value>
```
this looks like:
```
if true === true; return true
else return false;
```
I think you could just do:
```
xValuesMatch = this.gridPosition[0][1] === this.gridPosition[1][1]
```
which should just evaluate to true or false.